clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp

```diff
@@ -1404,8 +1404,21 @@ static Error UnbundleFiles() {
     auto Output = Worklist.find(CurTriple);
     // The file may have more bundles for other targets, that we don't care
     // about. Therefore, move on to the next triple
-    if (Output == Worklist.end())
-      continue;
+    if (Output == Worklist.end()) {
+      // FIXME: temporary solution for supporting binaries produced by old
+      // versions of SYCL toolchain. Old versions used triples with 'sycldevice'
+      // environment component of the triple, whereas new toolchain use
+      // 'unknown' value for that triple component. Here we assume that if we
+      // are looking for a bundle for sycl offload kind, for the same target
+      // triple there might be either one with 'sycldevice' environment or with
+      // 'unknown' environment, but not both. So we will look for any of these
+      // two variants.
+      Output =
+          Worklist.find(CurTriple.drop_back(11 /*length of "-sycldevice"*/));
```

Comment:
Is triple always larger than 11 chars?

Reply:
Good point. I refactored the code to avoid this problem.